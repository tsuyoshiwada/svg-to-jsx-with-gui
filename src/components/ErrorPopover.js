import React, { Component } from 'react';
import CSSTransitionGroup from 'react-transition-group/CSSTransitionGroup';
import styled from 'styled-components';
import { palette, easings } from '../styles';
import { Warning } from './icons/';
import firstChild from '../utils/firstChild';


const Popover = styled.div`
  display: flex;
  align-items: center;
  position: fixed;
  bottom: 80px;
  left: 50%;
  z-index: 3000;
  background: ${palette.danger};
  border-radius: 2px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.16);
  color: #fff;
  font-size: 0.75rem;
  font-weight: bold;
  transform: translate(-50%, 0);
  transition-property: opacity, visibility, transform;
  transition-timing-function: ${easings.easeOutQuart};

  & > div {
    flex: 1 1 auto;

    &:nth-child(1) {
      padding: 1em;
      background: ${palette.dangerDark};
    }

    &:nth-child(2) {
      padding: 1em 1.5em;
    }
  }

  &.popover-enter {
    opacity: 0;
    visibility: hidden;
    transform: translate(-50%, 10px);
    transition-duration: 320ms;
  }

  &.popover-enter.popover-enter-active {
    opacity: 1;
    visibility: visible;
    transform: translate(-50%, 0);
  }

  &.popover-leave {
    opacity: 1;
    visibility: visible;
    transform: translate(-50%, 0);
    transition-duration: 120ms;
  }

  &.popover-leave.popover-leave-active {
    opacity: 0;
    visibility: hidden;
    transform: translate(-50%, 5px);
  }
`;

export default class ErrorPopover extends Component {
  render() {
    const { children } = this.props;

    return (
      <CSSTransitionGroup
        component={firstChild}
        transitionName="popover"
        transitionEnterTimeout={320}
        transitionLeaveTimeout={120}
      >
        {children &&
          <Popover
            onClick={this.handleClick}
          >
            <div>
              <Warning width={16} height={16} />
            </div>
            <div>
              {children}
            </div>
          </Popover>
        }
      </CSSTransitionGroup>
    );
  }
}
