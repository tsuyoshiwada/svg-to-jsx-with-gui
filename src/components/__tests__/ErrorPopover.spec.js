import React from 'react';
import { shallow } from 'enzyme';
import ErrorPopover from '../ErrorPopover';


describe('components#<ErrorPopover />', () => {
  test('Should be render', () => {
    const wrapper = shallow(<ErrorPopover />);
    expect(wrapper).toMatchSnapshot();
  });
});
