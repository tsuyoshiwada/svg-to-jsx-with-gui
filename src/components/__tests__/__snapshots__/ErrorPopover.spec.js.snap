// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components#<ErrorPopover /> Should be render 1`] = `
<CSSTransitionGroup
  component={[Function]}
  transitionAppear={false}
  transitionEnter={true}
  transitionEnterTimeout={320}
  transitionLeave={true}
  transitionLeaveTimeout={120}
  transitionName="popover"
/>
`;
